kythe/go/platform/kzip/kzip.go

```diff
@@ -148,9 +191,16 @@ func readUnit(digest string, f *zip.File) (*Unit, error) {
 		return nil, err
 	}
 	defer rc.Close()
-
 	var msg apb.IndexedCompilation
-	if err := jsonpb.Unmarshal(rc, &msg); err != nil {
+	if strings.HasPrefix(f.Name, "root/units.pb") {
+		rec := make([]byte, f.UncompressedSize64)
+		if _, err = io.ReadFull(rc, rec); err != nil {
+			return nil, err
+		}
+		if err := proto.Unmarshal(rec, &msg); err != nil {
+			return nil, fmt.Errorf("Error unmarshaling for %s: %s", digest, err)
```

Comment:
Lowercase first letter of error strings: `return nil, fmt.Errorf("error unmarshaling for digest %q: %v", digest, err)`